tools/ci_connector_ops/ci_connector_ops/pipelines/contexts.py

```diff
@@ -402,24 +416,20 @@ async def __aexit__(
         local_reports_path_root = "tools/ci_connector_ops/pipeline_reports/"
         connector_name = self.report.pipeline_context.connector.technical_name
         connector_version = self.report.pipeline_context.connector.version
-        git_revision = self.report.pipeline_context.git_revision
-        git_branch = self.report.pipeline_context.git_branch.replace("/", "_")
-        suffix = f"{connector_name}/{git_branch}/{connector_version}/{git_revision}.json"
-        local_report_path = Path(local_reports_path_root + suffix)
-        await local_report_path.parents[0].mkdir(parents=True, exist_ok=True)
-        await local_report_path.write_text(self.report.to_json())
-        if self.report.should_be_saved:
-            s3_key = self.s3_report_key + suffix
-            report_upload_exit_code = await remote_storage.upload_to_s3(
-                self.dagger_client, str(local_report_path), s3_key, os.environ["TEST_REPORTS_BUCKET_NAME"]
-            )
-            if report_upload_exit_code != 0:
-                self.logger.error("Uploading the report to S3 failed.")
+
+        suffix = f"{connector_name}/{connector_version}/output.json"
+        file_path_key = f"{self.report_output_prefix}/{suffix}"
+
+        await self.report.save(file_path_key)
```

Comment:
```suggestion
        await self.report.share(file_path_key)
```
What do you think about exposing a "mega" function on the report side that:
* Write report to remote storage if ci
* Write report to local file system if local
* Post a comment on the PR if ci
* Print out the report to the console
* Send a slack message if ci - on publish only but could be used for connector health reports.

It makes me think that we could revamp the connector health report logic on slack and just report to slack at runtime, one message per connector. 

We'd totally defer what needs to be reported and when  to the Report class.

Let me know what you think about orchestrating slack messages from Dagger instead of Dagster. The dagger pipeline could also retrieve the previous build statuses from GCS.
I believe this approach would totally remove the need for a "complete.json" file and downstream aggretion and processing on Dagster. The connector badge could also be uploaded from Dagger... Looks like a nice shortcut ATM. Wdyt?

Reply:
@alafanechere I like this direction!

The only thing we loose is history
1. We wont have ✅❌✅✅✅❌
2. Our badges and connector specific status report history looking back

Reply:
We have timestamp on the key prefix, so we could `ls` the bucket with the prefix, order results by timestamp and pick the last 6 results.
But I believed this is a logic that's better suited for aggregation usecases in Dagster. And I like to have one big artifact produced daily that persists all our connectors tests results.
I'll let you decide if this is a beneficial shortcut compared to a Dagster job implementation that we could do later.